dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -288,154 +278,408 @@ def read_metadata(
+                    )
+
+            if file_row_groups[fpath]:
+                file_row_groups[fpath].append(file_row_groups[fpath][-1] + 1)
             else:
-                # Case (3C)
-                # We cannot pass a specific file/part path
-                file_path = paths
-                i_path = i
-
-            # Strip down pf object
-            if pf_deps and pf_deps != "tuple":
-                # Case (2)
-                for col in part.columns:
-                    col.meta_data.statistics = None
-                    col.meta_data.encoding_stats = None
-
-            # Final definition of "piece" and "pf" for this output partition
-            piece = i_path if pf_deps else part
-            pf_piece = (file_path, gather_statistics) if pf_deps == "tuple" else pf_deps
-            part_item = {
-                "piece": piece,
-                "kwargs": {"pf": pf_piece},
-            }
-            parts.append(part_item)
+                file_row_groups[fpath].append(0)
+
+            # Store mapping of local row-group to global row_group
+            global_lookup[(fpath, file_row_groups[fpath][-1])] = rg
+
+            if gather_statistics:
+                if single_rg_parts:
+                    s = {
+                        "file_path_0": fpath,
+                        "num-rows": row_group.num_rows,
+                        "total_byte_size": row_group.total_byte_size,
+                        "columns": [],
+                    }
+                else:
+                    s = {
+                        "num-rows": row_group.num_rows,
+                        "total_byte_size": row_group.total_byte_size,
+                    }
+                cstats = []
+                for name, i in stat_col_indices.items():
+                    column = row_group.columns[i]
+                    col = column.meta_data.path_in_schema[0]
+                    if column.meta_data.statistics:
+                        cmin = None
+                        cmax = None
+                        if pf.statistics["min"][col][0] is not None:
+                            cmin = pf.statistics["min"][col][rg]
+                            cmax = pf.statistics["max"][col][rg]
+                        elif dtypes[col] == "object":
+                            cmin = column.meta_data.statistics.min_value
+                            cmax = column.meta_data.statistics.max_value
+                            if isinstance(cmin, (bytes, bytearray)):
+                                cmin = cmin.decode("utf-8")
+                                cmax = cmax.decode("utf-8")
+                        if isinstance(cmin, np.datetime64):
+                            tz = getattr(dtypes[col], "tz", None)
+                            cmin = pd.Timestamp(cmin, tz=tz)
+                            cmax = pd.Timestamp(cmax, tz=tz)
+                        cnull = column.meta_data.statistics.null_count
```

Comment:
This is probably only useful for skipping a row-group where all the values are null

Reply:
>This is probably only useful for skipping a row-group where all the values are null

Right - That was the original reason the null count was included.  However, until we can confirm that such an optimization is valuable, it is probably makes the most sense to remove all null-count tracking from our statistics to avoid the necessary overhead.  I'm removing it - let me know if you have any objections :)